python/ray/data/tests/test_dataset.py

```diff
@@ -86,6 +88,54 @@ def count(s):
     assert not all(c == 6 for c in r2), r2
 
 
+@pytest.mark.parametrize(
+    "block_sizes,num_splits",
+    [
```

Comment:
given how delicate the algorithm is, can we do two more things:
1. not only verify the size of the block but also very the content of the blocks? This could be achieved by ensure each item is unique.
2. add some randomized tests.